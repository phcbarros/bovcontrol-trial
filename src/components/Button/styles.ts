import styled from 'styled-components/native'

export const Container = styled.TouchableOpacity`
  width: 100%;
  height: 56px;
  background-color: #8257e5;
  justify-content: center;
  align-items: center;
`

export const Title = styled.Text`
  color: #fff;
  font-size: 16px;
  font-weight: bold;
`
