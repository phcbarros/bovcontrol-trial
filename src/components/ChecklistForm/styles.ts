import styled from 'styled-components/native'

export const Container = styled.ScrollView.attrs({
  showsVerticalScrollIndicator: false,
  contentContainerStyle: {paddingBottom: 40, justifyContent: 'flex-start'},
})``

export const Form = styled.View`
  width: 100%;
  background-color: ${({theme}) => theme.COLORS.backgroundPrimary};
  padding: 20px;
  gap: 16px;
`
