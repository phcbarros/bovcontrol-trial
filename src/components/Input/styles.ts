import styled from 'styled-components/native'

export const Container = styled.View``

export const Group = styled.View`
  width: 100%;
  height: 56px;
  background-color: #fff;
  flex-direction: row;
  align-items: center;
  overflow: hidden;
`

export const Icon = styled.View`
  width: 56px;
  height: 56px;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  border-right-width: 3px;
  border-right-color: #f4f5f6;
`
export const Control = styled.TextInput`
  flex: 1;
  padding-left: 16px;
  font-size: 16px;
`

export const Error = styled.Text`
  font-size: 14px;
  color: #dc1637;
  margin-top: 8px;
`
