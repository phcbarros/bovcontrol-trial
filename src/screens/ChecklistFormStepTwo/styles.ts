import styled from 'styled-components/native'

export const Container = styled.View`
  flex: 1;
  background-color: ${({theme}) => theme.colors.backgroundPrimary};
  padding: 20px;
  justify-content: flex-start;
  align-items: center;
  gap: 16px;
`

export const Title = styled.Text`
  font-size: 24px;
  font-weight: bold;
  color: ${({theme}) => theme.colors.title};
  margin-top: 20px;
`
