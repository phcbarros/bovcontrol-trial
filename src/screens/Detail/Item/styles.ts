import styled from 'styled-components/native'

export const Container = styled.View`
  padding: 10px;
  border-radius: 5px;
  background-color: ${({theme}) => theme.colors.backgroundSecondary};
  margin-bottom: 10px;
`

export const Label = styled.Text`
  font-size: 16px;
  margin-bottom: 5px;
  font-weight: bold;
  color: ${({theme}) => theme.colors.title};
`

export const Value = styled(Label)`
  font-weight: normal;
`
