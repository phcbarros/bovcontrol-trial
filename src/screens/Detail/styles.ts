import styled from 'styled-components/native'

export const Container = styled.View`
  flex: 1;
  margin: 20px;
  gap: 8px;
  background-color: ${({theme}) => theme.colors.backgroundPrimary};
`

export const Scroll = styled.ScrollView.attrs({
  showsVerticalScrollIndicator: false,
  contentContainerStyle: {paddingBottom: 74},
})`
  flex: 1;
`

export const Item = styled.View`
  padding: 10px;
  border-radius: 5px;
  background-color: ${({theme}) => theme.colors.backgroundSecondary};
  margin-bottom: 10px;
`

export const Label = styled.Text`
  font-size: 16px;
  margin-bottom: 5px;
  font-weight: bold;
  color: ${({theme}) => theme.colors.title};
`

export const Text = styled(Label)`
  font-weight: normal;
`
