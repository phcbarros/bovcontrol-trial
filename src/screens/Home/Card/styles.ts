import styled from 'styled-components/native'

export const Container = styled.TouchableOpacity`
  background-color: ${({theme}) => theme.colors.backgroundPrimary};
  padding: 10px;
  border-radius: 5px;
  margin-bottom: 10px;
`

export const CardTitle = styled.Text`
  font-size: 16px;
  margin-bottom: 5px;
  font-weight: bold;
  color: ${({theme}) => theme.colors.title};
`

export const CardContent = styled.View``

export const Content = styled.View`
  flex-direction: row;
  justify-content: space-between;
`

export const Label = styled.Text`
  font-size: 14px;
  font-weight: bold;
  color: ${({theme}) => theme.colors.text};
`

export const Text = styled(Label)`
  color: ${({theme}) => theme.colors.textDetail};
`
