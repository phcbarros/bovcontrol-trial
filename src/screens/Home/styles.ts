import {FlatList} from 'react-native'
import styled from 'styled-components/native'
import {Checklist} from '../../types/checklist'

type ChecklistProps = Pick<Checklist, '_id' | 'from' | 'created_at' | 'farmer'>

export const Container = styled.View`
  flex: 1;
  width: 100%;
  background-color: ${({theme}) => theme.COLORS.backgroundSecondary};
  align-items: center;
  justify-content: center;
  gap: 10px;
  padding: 20px;
`

export const List = styled(FlatList<ChecklistProps>).attrs({
  showsVerticalScrollIndicator: false,
})`
  width: 100%;
  margin-bottom: 20px;
`
