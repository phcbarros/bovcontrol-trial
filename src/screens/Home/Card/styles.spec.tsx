import React from 'react'
import {render} from '@testing-library/react-native'
import 'jest-styled-components'
import 'jest-styled-components/native'

import {Container, CardTitle, Content, Label, Text} from './styles'
import {ThemeProvider} from 'styled-components/native'
import theme from '../../../styles/theme'

describe('Card.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const componente = render(<Container />, {
      wrapper: ({children}) => (
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      ),
    }).toJSON()

    expect(componente).toMatchSnapshot()
  })

  test('CardTitle DEVE ser igual ao snapshot', () => {
    const componente = render(<CardTitle />, {
      wrapper: ({children}) => (
        <ThemeProvider theme={{...theme}}>{children}</ThemeProvider>
      ),
    }).toJSON()

    expect(componente).toMatchSnapshot()
  })

  test('Content DEVE ser igual ao snapshot', () => {
    const componente = render(<Content />, {
      wrapper: ({children}) => (
        <ThemeProvider theme={{...theme}}>{children}</ThemeProvider>
      ),
    }).toJSON()

    expect(componente).toMatchSnapshot()
  })

  test('Label DEVE ser igual ao snapshot', () => {
    const componente = render(<Label />, {
      wrapper: ({children}) => (
        <ThemeProvider theme={{...theme}}>{children}</ThemeProvider>
      ),
    }).toJSON()

    expect(componente).toMatchSnapshot()
  })

  test('Text DEVE ser igual ao snapshot', () => {
    const componente = render(<Text />, {
      wrapper: ({children}) => (
        <ThemeProvider theme={{...theme}}>{children}</ThemeProvider>
      ),
    }).toJSON()

    expect(componente).toMatchSnapshot()
  })
})
