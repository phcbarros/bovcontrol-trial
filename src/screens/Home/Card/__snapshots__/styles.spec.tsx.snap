// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.styles CardTitle DEVE ser igual ao snapshot 1`] = `
<Text
  style={
    {
      "color": "#47474D",
      "fontSize": 16,
      "fontWeight": "bold",
      "marginBottom": 5,
    }
  }
/>
`;

exports[`Card.styles Container DEVE ser igual ao snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#F4F5F6",
      "borderBottomLeftRadius": 5,
      "borderBottomRightRadius": 5,
      "borderTopLeftRadius": 5,
      "borderTopRightRadius": 5,
      "marginBottom": 10,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
/>
`;

exports[`Card.styles Content DEVE ser igual ao snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
/>
`;

exports[`Card.styles Label DEVE ser igual ao snapshot 1`] = `
<Text
  style={
    {
      "color": "#7A7A80",
      "fontSize": 14,
      "fontWeight": "bold",
    }
  }
/>
`;

exports[`Card.styles Text DEVE ser igual ao snapshot 1`] = `
<Text
  style={
    {
      "color": "#AEAEB3",
      "fontSize": 14,
      "fontWeight": "bold",
    }
  }
/>
`;
